### PR TITLE
Replaced Future with CancelableFuture

### PR DESCRIPTION
**Pull request by aoprisan**
I have replaced all Futures with Cancelable Future and renamed the methods returning Future to their original name with 'F' appended.

### CODE REVIEW

File: src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala [hunk 1]
```diff
@@ -354,7 +354,16 @@ class SpyMemcachedIntegration(cf: ConnectionFactory, addrs: Seq[InetSocketAddres
...
+    val operationCancelable = Cancelable(() => {
+      try {
+        if (!op.isCancelled)
+          op.cancel()
+      } catch {
+        case NonFatal(_) =>
+      }
+    })
+
     val cancelable = scheduler.scheduleOnce(atMost) {
```
Comment (alexandru) [inline src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala]: This cancelable should be canceled as well in your `operationCancelable`.

File: src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala [hunk 2]
```diff
@@ -354,7 +354,16 @@ class SpyMemcachedIntegration(cf: ConnectionFactory, addrs: Seq[InetSocketAddres
...
   }
 
-  protected final def prepareFuture[T](key: String, op: Operation, promise: Promise[Result[T]], atMost: FiniteDuration)(implicit ec: ExecutionContext): Future[Result[T]] = {
+  protected final def prepareFuture[T](key: String, op: Operation, promise: Promise[Result[T]], atMost: FiniteDuration)(implicit ec: ExecutionContext): CancelableFuture[Result[T]] = {
+    val operationCancelable = Cancelable(() => {
+      try {
+        if (!op.isCancelled)
+          op.cancel()
+      } catch {
+        case NonFatal(_) =>
```
Comment (alexandru) [inline src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala]: Catching errors silently isn't a good thing to do. Even if I did it myself in the past, it was a bad thing to do :-) 

Report the error by means of `ec.reportError(ex)`, which should log it.

File: src/main/scala/shade/memcached/Memcached.scala [hunk 3]
```diff
@@ -15,31 +18,31 @@ trait Memcached extends java.io.Closeable {
...
-    Await.result(add(key, value, exp), Duration.Inf)
+    Await.result(addF(key, value, exp), Duration.Inf)
 
   /**
    * Sets a (key, value) in the cache store.
    *
    * The expiry time can be Duration.Inf (infinite duration).
    */
-  def set[T](key: String, value: T, exp: Duration)(implicit codec: Codec[T]): Future[Unit]
+  def setF[T](key: String, value: T, exp: Duration)(implicit codec: Codec[T]): CancelableFuture[Unit]
```
Comment (alexandru) [inline src/main/scala/shade/memcached/Memcached.scala]: Dude, I just looked at the Monix implementation and the `F` suffix convention is used for something else (it's for operations that keep the original context, like `foldLeftF` which still returns an `Observable`).  We'll use the `L` suffix for `Task`, but for `Future` return types we haven't used any convention and we shouldn't do it here.

Sorry for misleading you :-P

File: src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala (hunk 2 above)
Comment (aoprisan) [inline src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala]: Will do.

File: src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala (hunk 1 above)
Comment (aoprisan) [inline src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala]: Makes sense.

File: build.sbt [hunk 4]
```diff
@@ -65,6 +65,7 @@ libraryDependencies ++= Seq(
   "net.spy"        % "spymemcached"     % "2.12.1",
   "org.slf4j"      % "slf4j-api"        % "1.7.21",
   "io.monix"       %% "monix-execution" % "2.1.1",
+  "io.monix"       %% "monix-eval" % "2.1.1",
```
Comment (alexandru) [inline build.sbt]: Dude, when something is aligned, keep the alignment :-)

**Pull request merged**